app/assets/javascripts/admin/spree/orders/shipments.js.erb

```diff
@@ -39,87 +39,79 @@ $(document).ready(function() {
   }
   $('[data-hook=admin_order_edit_form] a.save-method').click(handle_shipping_method_save);
 
-  //handle tracking edit click
+  //handle tracking info edit/delete
+
+  // Show the input field to edit the tracking info
+  // And hide the input field when cancel is clicked
   $('a.edit-tracking').click(toggleTrackingEdit);
   $('a.cancel-tracking').click(toggleTrackingEdit);
 
+  saveTrackingInfo = function(){
+    let shipmentNumber = $(this).data('shipment-number');
+    let tracking = document.getElementById('tracking').value
+
+    makeApiCall(getTrackingUrl(shipmentNumber), { shipment: { tracking: tracking } } )
+  }
+
+  deleteTrackingInfo = function(){
+    let shipmentNumber = $(this).data('shipment-number');
+    let tracking = ''
+
+    confirmDelete(getTrackingUrl(shipmentNumber), { shipment: { tracking: tracking } })
+  }
+
+  getTrackingUrl = function(shipmentNumber){
```

Comment:
I would have called it `trackingUrl` to not confuse it with the HTTP GET we use here, too. We are simply assembling a string without "getting" it from somewhere.

Reply:
I understand what you mean and I agree so I changed it to `trackingUrl`